### PR TITLE
Improve position history data for sync mode

### PR DESCRIPTION
**Pull request by ZIMkaRU**

### CODE REVIEW

File: workers/loc.api/sync/data.inserter/halpers.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+  )
+
+  if (
+    !Array.isArray(trades) ||
+    trades.length === 0
+  ) {
+    return null
+  }
+  if (
```
Comment (ezewer) [inline workers/loc.api/sync/data.inserter/halpers.js:33]: Why is this if?

File: workers/loc.api/sync/data.inserter/api.middleware.handler.after.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          ...position,
+          closePrice,
+          pl: null,
+          plPerc: null
+        })
+
+        continue
+      }
+
+      const pl = closePrice - basePrice
```
Comment (ezewer) [inline workers/loc.api/sync/data.inserter/api.middleware.handler.after.js]: const pl = (closePrice - basePrice)  * traded amount

File: workers/loc.api/sync/data.inserter/api.middleware.handler.after.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      })
+    }
+
+    return {
+      ...apiRes,
+      res
+    }
+  }
+
+  _getPublicTrades (args, apiRes) {
```
Comment (ezewer) [inline workers/loc.api/sync/data.inserter/api.middleware.handler.after.js:86]: why is this added?

**Review by ezewer — COMMENTED** [review]
On general looks good, two questions and need to confirm something before continuing

File: workers/loc.api/sync/data.inserter/api.middleware.handler.after.js (hunk 2 above)
Comment (ZIMkaRU) [reply workers/loc.api/sync/data.inserter/api.middleware.handler.after.js]: It calculates as you propose
screenshot from 2019-02-13 10-01-20

and this result calculation, it seems as you provided
screenshot from 2019-02-13 11-15-33

Let me know if it need to change

File: workers/loc.api/sync/data.inserter/api.middleware.handler.after.js (hunk 3 above)
Comment (ZIMkaRU) [reply workers/loc.api/sync/data.inserter/api.middleware.handler.after.js:86]: That has moved from this place:
https://github.com/bitfinexcom/bfx-report/blob/master/workers/loc.api/sync/data.inserter.js#L759
Since it does the same job (it completes data from api_v2 for synchronization)
Let's keep similar things in one place

File: workers/loc.api/sync/data.inserter/halpers.js (hunk 1 above)
Comment (ZIMkaRU) [reply workers/loc.api/sync/data.inserter/halpers.js:33]: For example, let's see the case of this `getPositionsHistory` respond:
```
[
...,
            {
                "symbol": "tBTCUSD",
                "status": "CLOSED",
                "amount": 0,
                "basePrice": 17869.84000001,
                "marginFunding": 0,
                "marginFundingType": 0,
                "pl": null,
                "plPerc": null,
                "liquidationPrice": null,
                "leverage": null,
                "placeholder": null,
                "id": 36222041,
                "mtsCreate": 1544631487000,
                "mtsUpdate": 1544637983000
            },
...
]
```
- for `"mtsUpdate": 1544637983000` requests trades from DB:
And in this case first row is closed trade (`orderID=1151367451`), rest no (`orderID=1151367423`)
screenshot from 2019-02-13 10-51-02

- but if do request trades for this `getPositionsHistory` respond (`"mtsUpdate": 1544623422000`):
```
[
...,
            {
                "symbol": "tBTCUSD",
                "status": "CLOSED",
                "amount": -0.88667444,
                "basePrice": 18787,
                "marginFunding": -4.31199977,
                "marginFundingType": 0,
                "pl": null,
                "plPerc": null,
                "liquidationPrice": null,
                "leverage": null,
                "placeholder": null,
                "id": 36222035,
                "mtsCreate": 1544606097000,
                "mtsUpdate": 1544623422000
            },
...
]
```
we do not have closed trade, and all `orderID=1151366801`
screenshot from 2019-02-13 10-59-33

It the way define availability closed trade
Let me know if you have a better way

File: workers/loc.api/sync/data.inserter/api.middleware.handler.after.js (hunk 2 above)
Comment (ezewer) [reply workers/loc.api/sync/data.inserter/api.middleware.handler.after.js]: yes, but this is a tricky case as the amount is 1, but if amount was 0.7 would be (closePrice - basePrice) * 0.7

File: workers/loc.api/sync/data.inserter/api.middleware.handler.after.js (hunk 3 above)
Comment (ezewer) [reply workers/loc.api/sync/data.inserter/api.middleware.handler.after.js:86]: ok

File: workers/loc.api/sync/data.inserter/halpers.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    id
+  }
+) => {
+  const trades = await dao.findInCollBy(
+    '_getTrades',
+    {
+      auth,
+      params: {
+        symbol,
+        end
```
Comment (ezewer) [inline workers/loc.api/sync/data.inserter/halpers.js]: As an idea we can add a limit as to not have a very big response that consumes memory

**Review by ezewer — APPROVED** [review]
